packages/authentication/modules/access/usePreAuth/index.ts

```diff
@@ -0,0 +1,60 @@
+import { ACCESS_COOKIE_NAME, REFRESH_COOKIE_NAME, TokenTypes } from '@baseapp-frontend/utils'
+
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import Cookies from 'js-cookie'
+
+import AuthApi from '../../../services/auth'
+import { USER_API_KEY } from '../../../services/user'
+import { useSimpleTokenUser } from '../../user'
+import { IUsePreAuth } from './types'
+import { isJWTResponse } from './utils'
+
+const usePreAuth = ({
+  token,
+  queryOptions = {},
+  tokenType = TokenTypes.jwt,
+  cookieName = ACCESS_COOKIE_NAME,
+  refreshCookieName = REFRESH_COOKIE_NAME,
+  ApiClass = AuthApi,
+}: IUsePreAuth) => {
+  const queryClient = useQueryClient()
+  const { refetch: refetchUser } = useSimpleTokenUser({ options: { enabled: false } })
+
+  const query = useQuery({
+    staleTime: Infinity, // makes cache never expire automatically
+    enabled: !!token,
+    useErrorBoundary: false,
+    retry: false,
+    ...queryOptions, // needs to be placed bellow all overridable options
+    queryKey: ['preAuth', token, tokenType],
+    queryFn: () => ApiClass.preAuth({ token }, tokenType),
+    onSuccess: (response) => {
+      if (isJWTResponse(tokenType, response)) {
+        Cookies.set(cookieName, response.access, {
+          secure: process.env.NODE_ENV === 'production',
+        })
+        Cookies.set(refreshCookieName, response.refresh, {
+          secure: process.env.NODE_ENV === 'production',
+        })
+      } else {
+        Cookies.set(cookieName, response.token, {
+          secure: process.env.NODE_ENV === 'production',
+        })
+      }
+      // by invalidating the cache we force a reload of /v1/users/me and the state used by useUser hook
+      queryClient.invalidateQueries(USER_API_KEY.getUser())
+      refetchUser()
```

Comment:
let's use `resetQueries` instead of invalidate + refetching as we do [here](https://github.com/silverlogic/baseapp-frontend/blob/master/packages/authentication/modules/access/useLogout/index.ts#L20)
we had an issue during the logout that by just invalidating it wouldn't clear the previous cache.
so I think just `resetQueries` would be enough here, and then whenever we enter an authenticated page (that requires a user to be fetched) it will be fetched by a [withAuthentication HOC](https://bitbucket.org/silverlogic/baseapp-frontend-template/src/6698a3a054281305b12f7c0af17da33c0952b29c/.app-templates/mui-project/modules/auth/withAuthentication/index.tsx#lines-37) or something similar